src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php

```diff
@@ -263,15 +265,23 @@ private function registerReceivers(ContainerBuilder $container, array $busIds):
             $receiverMapping[$id] = new Reference($id);
 
             foreach ($tags as $tag) {
+                $receiverPriority[$id] = max($tag['priority'] ?? 0, $receiverPriority[$id] ?? PHP_INT_MIN);
```

Comment:
Does it happen that a receiver service has multiple `messenger.receiver` tags? If that never happens then the "take the max priority of all added tags" logic can be simplified to a single easier to read `$receiverPriority[$id] = 0`.

Based on the current `FrameworkExtension` code it looked like the answer would be "no, that never happens" but I didn't know whether there were other edge-cases.

Reply:
why not using PriorityTaggedServiceTrait::findAndSortTaggedServices instead?